internal/temporal/registration.go

```diff
@@ -1,28 +1,81 @@
 package temporal
 
 import (
+	"fmt"
+
 	"github.com/flexprice/flexprice/internal/service"
 	"github.com/flexprice/flexprice/internal/temporal/activities"
+	temporalService "github.com/flexprice/flexprice/internal/temporal/service"
 	"github.com/flexprice/flexprice/internal/temporal/workflows"
-	"go.temporal.io/sdk/worker"
+	"github.com/flexprice/flexprice/internal/types"
 )
 
-// RegisterWorkflowsAndActivities registers all workflows and activities with a Temporal worker.
-func RegisterWorkflowsAndActivities(w worker.Worker, params service.ServiceParams) {
-
-	// workflows - using function references
-	w.RegisterWorkflow(workflows.CronBillingWorkflow)      // "CronBillingWorkflow"
-	w.RegisterWorkflow(workflows.CalculateChargesWorkflow) // "CalculateChargesWorkflow"
-	w.RegisterWorkflow(workflows.PriceSyncWorkflow)        // "PriceSyncWorkflow"
+// WorkerConfig defines the configuration for a specific task queue worker
+type WorkerConfig struct {
+	TaskQueue  types.TemporalTaskQueue
+	Workflows  []interface{}
+	Activities []interface{}
+}
 
-	// activities - properly instantiate with dependencies
+// RegisterWorkflowsAndActivities registers all workflows and activities with the temporal service
+func RegisterWorkflowsAndActivities(temporalService temporalService.TemporalService, params service.ServiceParams) error {
+	// Create activity instances with dependencies
 	planService := service.NewPlanService(params)
 	planActivities := activities.NewPlanActivities(planService)
-	billingActivities := &activities.BillingActivities{}
 
-	// Register activities with explicit method names
-	w.RegisterActivity(planActivities.SyncPlanPrices)       // "SyncPlanPrices"
-	w.RegisterActivity(billingActivities.FetchDataActivity) // "FetchDataActivity"
-	w.RegisterActivity(billingActivities.CalculateActivity) // "CalculateActivity"
+	taskService := service.NewTaskService(params)
+	taskActivities := activities.NewTaskActivities(taskService)
+
+	// Get all task queues and register workflows/activities for each
+	for _, taskQueue := range types.GetAllTaskQueues() {
+		config := buildWorkerConfig(taskQueue, planActivities, taskActivities)
+		if err := registerWorker(temporalService, config); err != nil {
+			return fmt.Errorf("failed to register worker for task queue %s: %w", taskQueue, err)
+		}
+	}
+
+	return nil
+}
+
+// buildWorkerConfig creates a worker configuration for a specific task queue
+func buildWorkerConfig(taskQueue types.TemporalTaskQueue, planActivities *activities.PlanActivities, taskActivities *activities.TaskActivities) WorkerConfig {
+	workflowsList := []interface{}{}
+	activitiesList := []interface{}{}
+
+	switch taskQueue {
+	case types.TemporalTaskQueueTask:
+		workflowsList = append(workflowsList, workflows.TaskProcessingWorkflow)
+		activitiesList = append(activitiesList, taskActivities.ProcessTask)
```

Comment:
in future there can be common activities list as well right? say send webhook can be an activity which is being used by myltiple workflows and will need to be explicitly defined here at a queue level atleast. Just confirming the behaviour here.